probe-rs-target/src/chip.rs

```diff
@@ -1,6 +1,42 @@
 use super::memory::MemoryRegion;
 use crate::{serialize::hex_option, CoreType};
 use serde::{Deserialize, Serialize};
+
+/// Represents a DAP scan chain element.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct ScanChainDap {
+    /// Unique name of the DAP
+    pub name: Option<String>,
+    /// Specifies the type of the DAP (e.g. ARMCS-DP)
+    pub dap_type: Option<String>,
+    /// Specifies the IR length of the DAP (default value: 4).
+    pub ir_len: Option<u8>,
+    /// List of protocols via which the DAP can be use. Possible values are SWD, JTAG, and cJTAG.
+    pub protocols: Option<Vec<String>>,
+    /// Specifies the SW-DPv2 TARGETSEL register value that selects this DAP in a serial-wire debug multi-drop system
+    pub targetsel: Option<u8>,
```

Comment:
We already have a field for this. Can we not reuse that one? Is there a way to unify the two? I think the only chip currently really using it is the rp2040 :)

Reply:
I tried to define the struct as below, but then I realized that `probe-rs-target` doesn't have dependencies on `probe-rs` only the other way around. Since `DpAddress`, `PortType`, and `WireProtocol` are defined in probe-rs, I didn't add that dependency to avoid a loop.

Instead of doing that, I just paired down the definition of `ScanChainElement`


```rust
pub struct ScanChainElement {
    /// Unique name of the DAP
    pub name: Option<String>,
    /// Specifies the type of the DAP (e.g. ARMCS-DP)
    pub element_type: Option<PortType>,
    /// Specifies the IR length of the DAP (default value: 4).
    pub ir_len: Option<u8>,
    /// List of protocols via which the DAP can be use. Possible values are SWD, JTAG, and cJTAG.
    pub protocols: Option<Vec<WireProtcol>>,
    /// Specifies the SW-DPv2 TARGETSEL register value that selects this DAP in a serial-wire debug multi-drop system
    pub targetsel: Option<DpAddress>,
}
```

Reply:
I think we would have to define the types in `probe-rs-target` and have the same type in `probe-rs` with a simple `From` conversion impl.
I think proper typing should be applied :) What do you think? :)

Reply:
Sounds good to me!